src/hotspot/share/classfile/classFileParser.cpp

```diff
@@ -882,27 +857,23 @@ void ClassFileParser::parse_interfaces(const ClassFileStream* const stream,
 
     // Check if there's any duplicates in interfaces
     ResourceMark rm(THREAD);
-    NameSigHash** interface_names = NEW_RESOURCE_ARRAY_IN_THREAD(THREAD,
-                                                                 NameSigHash*,
-                                                                 HASH_ROW_SIZE);
-    initialize_hashtable(interface_names);
-    bool dup = false;
-    const Symbol* name = nullptr;
+    // Set containing interface names
+    NameSigHashtable* interface_names = new NameSigHashtable();
+    bool dup = true;
+    NameSigHash* interface_name = nullptr;
     {
       debug_only(NoSafepointVerifier nsv;)
-      for (index = 0; index < itfs_len; index++) {
+      for (index = 0; (index < itfs_len) && dup; index++) {
         const InstanceKlass* const k = _local_interfaces->at(index);
-        name = k->name();
+        interface_name = new NameSigHash();
```

Comment:
You could use a local NameSigHash and copy for the values like:
  name_sig_hash = NameSigHash(k->name(), nullptr);
Then the hashtable key could be a non-pointer, saving a pointer per entry and may be more efficient.